lib/Parse/ParseExpr.cpp

```diff
@@ -3218,6 +3219,9 @@ ParserResult<Expr> Parser::parseExprCollection() {
                                 *this, LSquareLoc,
                                 StructureMarkerKind::OpenSquare);
 
+  while (Tok.is(tok::pound_sourceLocation))
+    parseLineDirective();
```

Comment:
I could but I’d really rather not. I was asked to do this by @dabrahams and it’s so intertwined with this PR I’d need to wait until this PR closes out before I could put it back in.

Reply:
I agree with @rintaro.  The changes to `#sourceLocation` are independent and should be in a separate PR.

> could not back out one or other PR independently.

Nothing prevents backing out the `#sourceLocation` change.  The fact that there is a one-way dependency is not a good enough reason to avoid splitting the changes.

Incidentally, the `#sourceLocation` change doesn't appear to be tested.

Reply:
I’ll take it out.

Reply:
Thanks!